types/chart.js/index.d.ts

```diff
@@ -275,6 +278,10 @@ declare namespace Chart {
         borderWidth?: number;
     }
 
+    interface ChartTooltipsStaticConfiguartion {
+        positioners: any;
```

Comment:
That’s a really vague definition, but it doesn’t seem like there’s much to work with…

Maybe something like this:

```ts
interface ChartTooltipsStaticConfiguration {
    positioners: {[mode: string]: ChartTooltipPositioner};
}

type ChartTooltipPositioner = (elements: any[], eventPosition: Point) => Point;

interface Point {
    x: number;
    y: number;
}
```

The type of `elements` (described as "Chart.Element[]") is completely undocumented, AFAICT, so think it’s better to leave as `any`. And the documentation example mentioning `this` also appears to be bogus.

Reply:
Yes you are right we could improve the definition.

When I dump one element we get following structure:

```javascript
var ChartElement =
    {
        hidden: false,
        _chart: { ... },
        _datasetIndex: 0,
        _index: 1,
        _model: {
            "x": 74.31506588976988,
            "y": 546,
            "skip": false,
            "radius": 4,
            "pointStyle": "circle",
            "backgroundColor": "rgb(21, 172, 82)",
            "borderColor": "rgba(0, 0, 0, 0.1)",
            "borderWidth": 0,
            "tension": 0,
            "hitRadius": 1
        },
        _start: {},
        _view: {
            "x": 110.31506588976988,
            "y": 546,
            "skip": false,
            "radius": 4,
            "pointStyle": "circle",
            "backgroundColor": "rgb(21, 172, 82)",
            "borderColor": "rgba(0, 0, 0, 0.1)",
            "borderWidth": 0,
            "tension": 0,
            "hitRadius": 1
        },
        _xScale: { ... },
        _xScale: { ... },
    };
```

The properties with ```{ ... }``` have circular connections so I did not dump them. The only property which look promising is ```hidden```. But I would also stick to ```any```.

I'll push this changes for review.